FEXCore/include/FEXCore/Utils/SignalScopeGuards.h

```diff
@@ -202,7 +202,6 @@ static auto MaskSignalsAndLockMutex(MutexType& mutex, uint64_t Mask = ~0ULL) {
   } scope_guard {ScopedSignalMasker {Mask}, LockType<MutexType> {mutex}};
   return scope_guard;
 #else
-  // TODO: Doesn't block signals which may or may not cause issues.
```

Comment:
Since Windows indeed doesn't need this, could you drop the interleaved `#ifdef`s and instead implement WIN32-specific variants of the functions at the bottom of the file?

The logic here is already dense enough as-is, no need to make it even more complicated by branching on platforms in each function. This would also make for a good place to comment that the names are misleading on WIN32.